frontend/src/Components/CommentComponent.tsx

```diff
@@ -77,17 +78,19 @@ export default function CommentComponent(props: CommentProps) {
         }
     };
 
-    const translate = () => {
+    const translate = async () => {
         if (translation) {
             setTranslation(undefined);
         } else {
             setTranslation(false);
-            api.postAPI.translate(props.comment.id, 'comment')
-                .then(res => setTranslation(res.html))
-                .catch(() => {
-                    setTranslation(undefined);
-                    toast.error('Не удалось перевести');
-                });
+            try {
+                const html = await googleTranslate(props.comment.content);
```

Comment:
Let's memoize this. Some people tend to switch between translation and original several times.